src/service/createUpdateIdentifierPersona.ts

```diff
@@ -0,0 +1,112 @@
+import NoModel from 'jscommons/dist/errors/NoModel';
+import * as promiseRetry from 'promise-retry';
+import Locked from '../errors/Locked';
+import CreateUpdateIdentifierPersonaOptions // tslint:disable-line:import-spacing
+  from '../serviceFactory/options/CreateUpdateIdentifierPersonaOptions';
+import CreateUpdateIdentifierPersonaResult // tslint:disable-line:import-spacing
+  from '../serviceFactory/results/CreateUpdateIdentifierPersonaResult';
+import Config from './Config';
+import service from './index';
+
+const create = (config: Config) =>
+  async ({organisation, ifi, personaName}: CreateUpdateIdentifierPersonaOptions):
+  Promise<CreateUpdateIdentifierPersonaResult> => {
+    const { identifier } = await config.repo.createIdentifier({
```

Comment:
We could just assume locked is true in the repo because we're creating without an identifier and then setIdentifierPersona below can assume that locked is false because we're setting the persona.

Reply:
This is true. Technically we don't need 'locked' at all. but I like to have it as it makes the logic more explicit.